### PR TITLE
JS Binding

### PR DESCRIPTION
**Pull request by vhavlena**
Support for JS binding building for Z3-Noodler. Successfully builds the binding including the WASM library. There is still some issue to be solved because when this binding is used in JS, the main method `solver.check()` does nothing.

### CODE REVIEW

File: src/math/lp/static_matrix.h [hunk 1]
```diff
@@ -79,7 +79,7 @@ class static_matrix
         ref(static_matrix & m, unsigned row, unsigned col):m_matrix(m), m_row(row), m_col(col) {}
         ref & operator=(T const & v) { m_matrix.set( m_row, m_col, v); return *this; }
 
-        ref operator=(ref & v) { m_matrix.set(m_row, m_col, v.m_matrix.get(v.m_row, v.m_col)); return *this; }
+        ref operator=(ref & v) { m_matrix.set(m_row, m_col, v.m_matrix.get_val(v.m_row, v.m_col)); return *this; }
```
Comment (jurajsic) [inline src/math/lp/static_matrix.h:82]: Is this some javascript shit that needed to be fixed?

Comment (vhavlena) [reply src/math/lp/static_matrix.h:82]: These are regular compiler errors. But they don't show up during a regular build (maybe it has something to do with the emscripten, which is used for the building or the fact that we are using a static library build).

**Review by jurajsic — APPROVED** [review]

File: src/math/lp/static_matrix.h (hunk 1 above)
Comment (jurajsic) [reply src/math/lp/static_matrix.h:82]: That's really weird. Is this file included anywhere? Maybe it is not used but emscripten takes everything, even files that are not used.

Comment (vhavlena) [reply src/math/lp/static_matrix.h:82]: Yes, I have a similar feeling. It is included somewhere but it does not mean it is really used. Maybe this was a part of a dead code.

**Pull request merged**